docs/appendices/release-notes/unreleased.rst

```diff
@@ -71,6 +71,9 @@ Fixes
 - Fixed an issue that caused the returned column names to be missing the
   subscripts when querying sub-columns of nested object arrays.
 
+- Fixed an issue that caused ``ClassCastException`` when accessing a sub-column
+  of a nested object array where the sub-column resolves to a nested array.
```

Comment:
Maybe add an example here to make it 100% clear for users

Reply:
Thank you, added an example.